tests/ops/api/test_ratelimit.py

```diff
@@ -0,0 +1,26 @@
+import pytest
+
+from fides.api.ops.api.v1.urn_registry import HEALTH
+from fides.ctl.core.config import get_config
+
+CONFIG = get_config()
+
+
+def test_requests_ratelimited(api_client, cache):
+    """
+    Asserts that incremental HTTP requests above the ratelimit threshold are
+    rebuffed from the API with a 429 response.
+    """
```

Comment:
There's a failure condition of this test where the `CONFIG.security.request_rate_limit` var is set so high that the test client can't send requests quickly enough to beat it — that shouldn't be possible with the default setting of `100/minute`.

I've tried to override the limit to something artificially small for the sake of a more reliable test with the following:
```
REQUEST_LIMIT = 1
DEFAULT_TEST_RATE_LIMIT = [f"{REQUEST_LIMIT}/minute"]


def configure_rate_limits(
    app: FastAPI,
    limits: List[str],
    key_prefix: str = CONFIG.security.rate_limit_prefix,
) -> FastAPI:
    limiter = Limiter(
        default_limits=limits,
        headers_enabled=True,
        key_prefix=key_prefix,
        key_func=get_remote_address,
        retry_after="http-date",
        storage_uri=CONFIG.redis.connection_url,
    )
    app.state.limiter = limiter
    app.add_exception_handler(RateLimitExceeded, _rate_limit_exceeded_handler)
    return app


@pytest.fixture(scope="function")
def rate_limited_api_client():
    """
    Yields a client used to make API requests with an artificially low
    ratelimit set.
    """
    rate_limited_app = configure_rate_limits(
        app=app,
        limits=DEFAULT_TEST_RATE_LIMIT,
    )
    with TestClient(rate_limited_app) as client:
        yield client
```
however, the test seems to hang indefinitely. I believe it needs to be done this way as the rate limit is set on the `FastAPI` application at runtime.

Reply:
Can you just clarify `There's a failure condition of this test`--this test is passing for me, so I just want to better understand where to focus

Reply:
@seanpreston if there are caveats, please also note them in the test docstring too for future developers' sake

It sounds like this test doesn't reliably pass because the client can't reliably overwhelm the 100 requests/rolling minute, and that changing the default at runtime doesn't work either

Reply:
This is me highlighting a known edge case, I've never seen this test fail due to the reason, but it theoretically could. Will add a note in the docstring.

Reply:
Ah, I see! Appreciated for calling it out

Reply:
I tried something similar that seemed to work to lower the rate to "2/minute" so we didn't have to make 100+ requests for this test 

```python

@pytest.fixture(scope="function")
def api_client_for_rate_limiting() -> Generator:
    """Return a client used to make API requests"""
    app.state.limiter = Limiter(
        default_limits=["2/minute"],
        headers_enabled=True,
        key_prefix=CONFIG.security.rate_limit_prefix,
        key_func=get_remote_address,
        retry_after="http-date",
        storage_uri=CONFIG.redis.connection_url,
    )
    with TestClient(app) as c:
        yield c
        app.state.limiter = Limiter(
            default_limits=[CONFIG.security.request_rate_limit],
            headers_enabled=True,
            key_prefix=CONFIG.security.rate_limit_prefix,
            key_func=get_remote_address,
            retry_after="http-date",
            storage_uri=CONFIG.redis.connection_url,
        )


def test_requests_ratelimited(api_client_for_rate_limiting, cache):
    """
    Asserts that incremental HTTP requests above the ratelimit threshold are
    rebuffed from the API with a 429 response.

    A theoretical failure condition exists in this test should the container
    running it not be able to execute 100 requests against the client in a
    one minute period.
    """
    for _ in range(0, 2):
        response = api_client_for_rate_limiting.get(HEALTH)
        assert response.status_code == 200

    response = api_client_for_rate_limiting.get(HEALTH)
    assert response.status_code == 429

    ratelimiter_cache_keys = [key for key in cache.keys() if key.startswith("LIMITER/")]
    for key in ratelimiter_cache_keys:
        # Depending on what requests have been stored previously, the ratelimtier will
        # store keys in the format `LIMITER/fides-/127.0.0.1//health/100/1/minute`
        assert key.startswith(f"LIMITER/{CONFIG.security.rate_limit_prefix}")
        # Reset the cache to not interere with any other tests
        cache.delete(key)
```